parts/k8s/kubernetescustomscript.sh

```diff
@@ -171,12 +170,18 @@ if ! $FULL_INSTALL_REQUIRED; then
     cleanUpContainerImages
 fi
 
+if [ -f /var/run/reboot-required ]; then
+    REBOOTREQUIRED=true
+else
+    REBOOTREQUIRED=false
+fi
+
 if $REBOOTREQUIRED; then
   echo 'reboot required, rebooting node in 1 minute'
-  /bin/bash -c "shutdown -r 1 &"
   if [[ $OS == $UBUNTU_OS_NAME ]]; then
-      aptmarkWALinuxAgent unhold &
+      aptmarkWALinuxAgent unhold
   fi
+  nohup /bin/bash -c "sleep 3 && reboot" &
```

Comment:
Does an `apt-mark hold` persist reboots? If it doesn't, we don't need to worry about the fact that we're potentially rebooting before that completes.